syntax = "proto3";

package SafeEntry;

// The greeting service definition.
service SafeEntry {
  rpc checkIn (CheckInRequest) returns (CheckInReply) {}
  rpc checkOut (CheckOutRequest) returns (CheckOutReply) {}
  rpc groupCheckIn (stream GroupCheckInRequest) returns (GroupCheckInReply) {}
  rpc groupCheckOut (stream GroupCheckOutRequest) returns (GroupCheckOutReply) {}
  rpc getLocation (LocationRequest) returns (stream LocationReply) {}
  rpc updateLocation (MOHRequest) returns (MOHReply) {}
}

// The request message containing the user's name, NRIC, location, and check in check out time 
message CheckInRequest {
  string name = 1;
  string id = 2;
  string location = 3;
  string check_in_time = 4;
}

// The response message containing the check in essage reply from server
message CheckInReply {
  string res = 1;
}

// The request message containing the user's name, NRIC, location, and check out time 
message CheckOutRequest {
  string name = 1;
  string id = 2;
  string location = 3;
  string check_out_time = 4;
}

// The response message containing the check out essage reply from server
message CheckOutReply {
  string res = 1;
}

// The request message containing the user's name, NRIC, location, and check in check out time 
message GroupCheckInRequest {
  string name = 1;
  string id = 2;
  string location = 3;
  string check_in_time = 4;
}

// The response message containing the greetings
message GroupCheckInReply {
  string res = 1;
  repeated GroupCheckInRequest request = 2;
}

// The request message containing the user's name, NRIC, location, and check in check out time 
message GroupCheckOutRequest {
  string name = 1;
  string id = 2;
  string location = 3;
  string check_out_time = 4;
}

// The response message containing the check out reply message 
message GroupCheckOutReply {
  string res = 1;
  repeated GroupCheckOutRequest request = 2;
}

// The request message containing the location names provided by MOH  
message MOHRequest {
  string location_name = 1;
  string visit_date = 2;
}

// The response message containing the reply message sent to MOH 
message MOHReply {
  string res = 1;
}


// The request message containing the location names provided by MOH  
message LocationRequest {
  string user_name = 1;
  string user_id = 2;
}

// The response message containing the reply message sent to MOH 
message LocationReply {
  string location_name = 1;
}
